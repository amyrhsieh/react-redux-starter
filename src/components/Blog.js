import React from 'react';
import { connect } from 'react-redux';
import { createEntry } from '../redux/entry';

export const Blog = ({entries, onAddEntryClick}) => (
  <div>
    <h2>Blog</h2>
    <ul>
      {entries.map(entry =>
        (<li key={entry.timestamp}>{entry.content}</li>)
      )}
    </ul>
    <button type="button" className="btn btn-outline-primary" onClick={() => onAddEntryClick()}>Add entry</button>
  </div>
);

const mapStateToProps = state => ({
  entries: state.entries
});

const mapDispatchToProps = dispatch => {
  return {
    onAddEntryClick: () => {
      dispatch(createEntry())
    }
  }
}

const VisibleBlog = connect(
  mapStateToProps,
  mapDispatchToProps
)(Blog);

export default VisibleBlog;
