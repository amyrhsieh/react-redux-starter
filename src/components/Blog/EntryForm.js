import React from 'react';
import { render } from 'react-dom';
import { Form, Field } from 'react-final-form';

import styles from './entryForm.scss';

const onSubmit = values => {
  window.alert(JSON.stringify(values, 0, 2));
}

const EntryForm = () => (
  <Form
    onSubmit={onSubmit}
    render={({ handleSubmit, form, submitting, pristine, values }) => (
      <form onSubmit={handleSubmit}>
        <div>
          <label className="formLabel">New Entry</label>
          <Field
            name="content"
            component="textarea"
            placeholder="Write your blog entry here"
          />
        </div>
        <div className="buttons">
          <button type="submit" disabled={submitting || pristine}>
            Submit
          </button>
          <button
            type="button"
            onClick={form.reset}
            disabled={submitting || pristine}
          >
            Reset
          </button>
        </div>
        <pre>{JSON.stringify(values, 0, 2)}</pre>
      </form>
    )}
  />
);

export default EntryForm;
