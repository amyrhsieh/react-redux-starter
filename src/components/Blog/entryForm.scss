.formLabel {
  font-weight: bold;
  display: block;
}

textarea {
  width: 550px;
  height: 100px;
}

button + button {
  margin-left: 10px;
}
