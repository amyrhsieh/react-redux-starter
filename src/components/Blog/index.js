import React from 'react';
import { connect } from 'react-redux';
import { createEntry } from 'Redux/entry';
import EntryForm from './EntryForm';

import styles from 'Styles/styles.scss';

export const Blog = ({entries, dispatchCreateEntry}) => (
  <div>
    <h2>Blog</h2>
    <p>
      A very simple blog demo that displays blog entries from the Redux store. Includes form (React Final Form) to add new entries.
    </p>
    <p className="bold">Blog Entries</p>
    <ul>
      {entries.map(entry =>
        (<li key={entry.timestamp}>{entry.content}</li>)
      )}
    </ul>
    <EntryForm onSubmit={dispatchCreateEntry} />
  </div>
);

const mapStateToProps = state => ({
  entries: state.entries
});

const mapDispatchToProps = dispatch => {
  return {
    dispatchCreateEntry: (values) => {
      const entry = {
        content: values.content,
        timestamp: Date.now()
      }
      dispatch(createEntry(entry));
    }
  }
}

const VisibleBlog = connect(
  mapStateToProps,
  mapDispatchToProps
)(Blog);

export default VisibleBlog;
