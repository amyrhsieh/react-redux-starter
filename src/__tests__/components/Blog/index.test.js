import React from 'react';
import { Blog } from 'Components/Blog';
import entries from 'Mocks/entriesMock';
import { shallow } from 'enzyme';

const dispatchCreateEntry = () => {return true;}

describe('Blog (Snapshot)', () => {
  it('Blog component renders Blog', () => {
    const wrapper = shallow(<Blog entries={entries} dispatchCreateEntry={dispatchCreateEntry} />);
    expect(wrapper).toMatchSnapshot();
  });
});
