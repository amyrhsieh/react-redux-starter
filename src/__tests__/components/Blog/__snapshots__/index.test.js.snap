// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog (Snapshot) Blog component renders Blog 1`] = `
<div>
  <h2>
    Blog
  </h2>
  <p>
    A very simple blog demo that displays blog entries from the Redux store. Includes form (React Final Form) to add new entries.
  </p>
  <p
    className="bold"
  >
    Blog Entries
  </p>
  <ul>
    <li
      key="1560427200"
    >
      Hello world! This is my first blog entry.
    </li>
    <li
      key="1560513600"
    >
      This is my second blog entry.
    </li>
  </ul>
  <EntryForm
    onSubmit={[Function]}
  />
</div>
`;
